base/base_tests/small_set_test.cpp

```diff
@@ -0,0 +1,74 @@
+#include "testing/testing.hpp"
+
+#include "base/small_set.hpp"
+
+#include <algorithm>
+#include <iterator>
+#include <vector>
+
+using namespace base;
+
+namespace
+{
+UNIT_TEST(SmallSet_Empty)
+{
+  SmallSet<0> set;
```

Comment:
It's up to you, but I prefer not to use value names which are equal to names from standard lib like set, map and so on.